spec/index.bs

```diff
@@ -4629,7 +4617,7 @@ and switched as one
 
 <script type=idl>
 dictionary GPUPipelineDescriptorBase : GPUObjectDescriptorBase {
-    GPUPipelineLayout layout;
+    required GPUPipelineLayoutRequirements layout;
```

Comment:
Should this allow `"auto"` as well to allow for fully-implicit pipeline layouts?

Reply:
This is a good point. As-is, this would appear to require developers to at least know how many bind groups are used by the shader and insert an `"auto"` for each of them. That's not terrible, you have to know how many bind groups there are or you're not going to be able to draw/dispatch with it, but it feels bulky for a scenario where I'm OK with the pipeline being fully implicit.

Reply:
I don't feel very strongly but don't think it's that necessary.
`layout: { bindGroupLayouts: ['auto', 'auto'] },` is not that much boilerplate (could maybe be shorter - we could mix it in for just `bindGroupLayouts: ['auto', 'auto'],` if we want). Pedagogically, it's nice that it gives you a very clear place to put your explicit layout as soon as you need it.